File: react-frontend/src/components/ListPage.js
import AddPlace from "./AddPlace";
import Weather from "./weather";
import React , { useState, useEffect }from "react";
import {Container,Row, Col,Button,Modal,ListGroup} from "react-bootstrap";

export default function  ListPage()  {
  const [listInfo, setlistInfo] = useState([]);
  const [listdata, setlistdata] = useState([]);
  var [latlon, setlatlon] = useState('');

  useEffect(() => {
      fetch('http://localhost:5000/getplaces',{
          credentials: "include"
        })
          .then(response => response.json())
          .then(object => setlistInfo(object))
      },[]);
     
  useEffect(() => {
      fetch('http://localhost:5000/getListInfo',{
         credentials: "include"
        })
         .then(response => response.json())
         .then(object => setlistdata(object))
      },[]);


    
        
    async function Setlat(vvv) {
      var da=vvv.name

      da=da.split(",")[0]
      da=da.split(" ")[0]
      const key= "5d50cb77a4d850371ce5a430e31c9b24"
      const response = await fetch("http://api.openweathermap.org/geo/1.0/direct?q="+da+"&limit=1"+"&appid="+key);
      const data = await response.json();
      setlatlon({"lat":data[0].lat,"lon":data[0].lon})
      if(latlon===''){ console.log("wait")}
     else{console.log(latlon)} }


  const ListGroupItem = (lgi, index) => {
 
      return ( 
              <ListGroup.Item  variant="default" key={index} style={{ textAlign: 'left', color: "black", background: "#1ca0f9" }} as="li">
                <Row><Col><h5>place name</h5></Col><Col><h5>start date</h5></Col><Col><h5>end date</h5></Col></Row>
                <Row><Col>{lgi.name}</Col><Col> {lgi.start_date}</Col><Col>{lgi.end_date}</Col></Row>
                  
                      <form action="http://localhost:5000/removeplace" method="post">
                      
                      <Button variant="danger" className="float-end" type="submit" name="place_id" value={lgi.id}>remove</Button>
                      {console.log("gggg")}
                      </form>
                     
                      <form action="http://localhost:5000/viewplace" method="post">
                      
                      <Button variant="success" className="float-end" type="submit" name="place_id" value={lgi.id}>view place</Button>
                     </form>

                   
              </ListGroup.Item>
             )   
  };
  

    return (

      <div className="auth-wrapper">
          <Row>
        <div className="auth-inner-left">
          <h1>{listdata.name}</h1>
          <hr class="my-4"></hr>
          <Row style={{'height': '298px', 'overflow':'auto'}}>

          <ListGroup defaultActiveKey="#link1" as="ol" numbered>
            
          {listInfo.map(ListGroupItem)}
          
        </ListGroup>
          </Row>
          <br />
          <Row>
         <AddPlace/>
          </Row>
          </div>
          <br />
          </Row>
        </div>

    );

}

File: flask-backend/endpoints/ListPage.py
from re import split
from flask import Blueprint, session, request, redirect, jsonify
from app import db
from models import Place, List
import requests

ListPage = Blueprint('ListPage',__name__)

@ListPage.route('/addplace', methods=['GET','POST'])
def addplace():
    if request.method == 'POST':
        user_id = session.get("user_id")
        list_id = session.get("list_id")
        place_name = request.form['place_name']
        start_date = request.form['start_date']
        end_date = request.form['end_date']
        lat_lon_dict = get_lat_lon(place_name)
        lat = lat_lon_dict["lat"]
        lon = lat_lon_dict["lon"]
        new_place = Place(user_id = user_id, list_id = list_id, name = place_name,start_date=start_date,end_date=end_date, lat = lat, lon = lon)
        db.session.add(new_place)
        db.session.commit()
        return redirect('http://localhost:3000/userpage/places')

@ListPage.route('/removeplace', methods=['GET','POST'])
def removeplace():
    if request.method == 'POST':
        user_id = session.get('user_id')
        list_id = session.get('list_id')
        id = request.form['place_id']
        place = Place.query.filter_by(user_id=user_id,list_id=list_id,id=id).first()
        db.session.delete(place)
        db.session.commit()
        return redirect('http://localhost:3000/userPage/places')

@ListPage.route('/getplaces', methods=['GET', 'POST'])
def getplaces():
    if request.method == 'GET':
        user_id = session.get('user_id')
        list_id = session.get('list_id')
        places = Place.query.filter_by(user_id=user_id,list_id=list_id).all()
        return jsonify(places)

@ListPage.route('/viewplace', methods=['GET','POST'])
def viewPlace():
    if request.method == 'POST':
        place_id = request.form['place_id']
        session['place_id'] = place_id
    return redirect('http://localhost:3000/UserPage/places/place')

@ListPage.route('/getListInfo', methods=['GET','POST'])
def getListInfo():
    if request.method == 'GET':
        user_id = session.get('user_id')
        list_id = session.get('list_id')
        list_info = List.query.filter_by(user_id=user_id,id=list_id).first()
    return jsonify(list_info)

def get_lat_lon(name):
    lat_lon_dict = {}
    name = name.split(",")[0]
    name = name.split("-")[0]
    api_key = '5d50cb77a4d850371ce5a430e31c9b24'
    api_url = "http://api.openweathermap.org/data/2.5/weather?q={}&units=metric&appid={}".format(name, api_key)
    data = requests.get(api_url)
    data = data.json()
    if data["cod"] == "404":
        name = name.split()[0]
        api_url = "http://api.openweathermap.org/data/2.5/weather?q={}&units=metric&appid={}".format(name, api_key)
        data = requests.get(api_url)
        data = data.json()
        if data["cod"] == "404":
            lat_lon_dict.update({"lat": 0})
            lat_lon_dict.update({"lon": 0})
            return lat_lon_dict
    lat_lon_dict.update({"lat": data["coord"]["lat"]})
    lat_lon_dict.update({"lon": data["coord"]["lon"]})
    return lat_lon_dict